test/integration/dbversion.test.js

```diff
@@ -0,0 +1,57 @@
+describe('dbVersion integration test', () => {
+  const dbVersion = require('../../server/dbversion')
+
+  async function truncateVersionTable () {
+    return dbVersion.umzug.storage.model.destroy({
+      where: {},
+      truncate: true
+    })
+  }
+
+  beforeEach(async () => {
+    await truncateVersionTable()
+  })
+
+  test('dbVersion gets a list of available migrations', () => {
+    expect(dbVersion.availableVersions.length).toBeGreaterThan(0)
+  })
+
+  test('dbVersion validates version in database', async () => {
+    await dbVersion.umzug.storage.model.upsert({ name: dbVersion.highestVersion })
+    await dbVersion.refreshCurrentDatabaseVersion()
+    expect(dbVersion.currentDatabaseVersion).toBe(dbVersion.highestVersion)
+  })
+
+  test('dbVersion reports a list of pending migrations', async () => {
+    const pending = await dbVersion.getPending()
+    expect(pending.length).toBeGreaterThan(0)
+  })
+
+  test('dbVersion reports an issue if there are pending migrations', async () => {
+    await expect(dbVersion.throwAnyErrors()).rejects.toThrow('Pending Database migrations exist')
+  })
+
+  test('dbVersion reports an issue if there are no database versions stored', async () => {
+    const oldFunction = dbVersion.getPending
+    dbVersion.getPending = jest.fn(() => { return [] })
+
+    await expect(dbVersion.throwAnyErrors()).rejects.toThrow('No database version could be found')
+
+    dbVersion.getPending = oldFunction
```

Comment:
There's a risk that the function isn't reset if the test fails, it might be better to reset the function in an afterEach in this test and the one below

Reply:
I've moved this to an afterEach now,